src/Process/Deploy/InstallUpdate/ConfigUpdate/SearchEngine.php

```diff
@@ -56,57 +47,69 @@ class SearchEngine implements ProcessInterface
      */
     private $searchEngineConfig;
 
+    /**
+     * @var ElasticSuite
+     */
+    private $elasticSuite;
+
     /**
      * @param LoggerInterface $logger
      * @param EnvWriter $envWriter
-     * @param EnvReader $envReader
      * @param SharedWriter $sharedWriter
-     * @param SharedReader $sharedReader
      * @param MagentoVersion $version
      * @param SearchEngineConfig $searchEngineConfig
+     * @param ElasticSuite $elasticSuite
      */
     public function __construct(
         LoggerInterface $logger,
         EnvWriter $envWriter,
-        EnvReader $envReader,
         SharedWriter $sharedWriter,
-        SharedReader $sharedReader,
         MagentoVersion $version,
-        SearchEngineConfig $searchEngineConfig
+        SearchEngineConfig $searchEngineConfig,
+        ElasticSuite $elasticSuite
     ) {
         $this->logger = $logger;
         $this->envWriter = $envWriter;
-        $this->envReader = $envReader;
         $this->sharedWriter = $sharedWriter;
-        $this->sharedReader = $sharedReader;
         $this->magentoVersion = $version;
         $this->searchEngineConfig = $searchEngineConfig;
+        $this->elasticSuite = $elasticSuite;
     }
 
     /**
      * Executes the process.
      *
      * @return void
+     * @throws ProcessException
      */
     public function execute()
     {
-        $this->logger->info('Updating search engine configuration.');
-
         $searchConfig = $this->searchEngineConfig->get();
 
+        $this->logger->info('Updating search engine configuration.');
         $this->logger->info('Set search engine to: ' . $searchConfig['engine']);
 
-        // 2.1.x requires search config to be written to the shared config file: MAGECLOUD-1317
-        if (!$this->magentoVersion->isGreaterOrEqual('2.2')) {
-            $config = $this->sharedReader->read();
-            $config['system']['default']['catalog']['search'] = $searchConfig;
-            $this->sharedWriter->create($config);
+        $config['system']['default']['catalog']['search'] = $searchConfig;
+
+        if ($this->elasticSuite->isAvailable()) {
+            $this->logger->info('Configuring ElasticSuite');
 
-            return;
+            $config['system']['default']['smile_elasticsuite_core_base_settings'] = $this->elasticSuite->get();
+        } elseif ($this->elasticSuite->isInstalled()) {
```

Comment:
This validation should be moved to the beginning of the deploy phase and it should fail the deployment for now, as Magento won't work with elasticsuite and without ES, even if engine was set as __mysql__ :(

Reply:
It is an edge case

Reply:
We need to cover this case. Customer should know that something went wrong and he should know the reason of it. We should not allow displaying successful deploy with broken store configs.

Reply:
If there is nor ElasticSearch - default engine will be mysql. Why it's bothering you?